docs/application/native/guides/text-input/capi-ui-gesture.md

```diff
@@ -0,0 +1,247 @@
+    {
+        hand_gesture_h gesture_h;
+        int ret;
+        ret = hand_gesture_create(&gesture_h);
+        if (HAND_GESTURE_ERROR_NONE != ret)
+            /* Error handling */
+    }
+    ```
+
+2. When you no longer need the Gesture library, destroy the Gesture handle using `hand_gesture_destroy()`:
+
+    ```c
+    void
+    destroy_gesture_handle(hand_gesture_h gesture_h)
+    {
+        int ret;
+        ret = hand_gesture_destroy(gesture_h); /* gesture_h is the Gesture handle */
+        if (HAND_GESTURE_ERROR_NONE != ret)
+            /* Error handling */
+    }
+    ```
+
+
+<a name="info_gesture"></a>
+## Get Gesture information
+
+You can get the following information about Gesture:
+
+1. Get the supported gesture types by checking whether the gesture type you want is supported or not.
+
+   According to the Gesture engine service, the supported gesture types are different.
+   The current Gesture engine service supports only one gesture type, `HAND_GESTURE_WRIST_UP`.
+   If the Gesture engine service is changed to support more gesture types, you can check more supported gesture types using `hand_gesture_is_supported_type()`.
+   The candidates of the gesture types are defined in the `gesture_common.h` header file as the enumerations of `hand_gesture_type_e`:
+
+    ```c
+    void
+    is_supported_gesture_type(hand_gesture_h gesture_h, hand_gesture_type_e gesture_type, bool* is_supported)
+    {
+        int ret;
+        ret = hand_gesture_is_supported_type(gesture_h, gesture_type, is_supported);
+        if (HAND_GESTURE_ERROR_NONE != ret)
+            /* Error handling */
+    }
+    ```
+
+2. Get the information of the Gesture engine service used in current.
+
+   You can get the application ID and name of the current Gesture engine service using `hand_gesture_get_engine_info()`:
+
+    ```c
+    void
+    get_info(hand_gesture_h gesture_h)
+    {
+        int ret;
+        char* engine_app_id;
+        char* engine_name;
+
+        ret = hand_gesture_get_engine_info(gesture_h, &engine_app_id, &engine_name);
+        if (HAND_GESTURE_ERROR_NONE != ret)
+            /* Error handling */
+    }
+    ```
+
+
+<a name="set_option"></a>
+## Set option
+
+You can select an option to detect hand gestures continuously or not. If you want to make the application detect hand gestures continuously, set the option as `HAND_GESTURE_OPTION_ALWAYS_ON` using `hand_gesture_set_option()`:
+
+```c
+void
+set_option(hand_gesture_h gesture_h)
+{
+    int ret;
+    ret = hand_gesture_set_option(gesture_h, HAND_GESTURE_OPTION_ALWAYS_ON);
+    if (HAND_GESTURE_ERROR_NONE != ret)
+        /* Error handling */
+}
+```
+
+
+<a name="callback"></a>
+## Set and unset callbacks
+
+The Gesture provides two types of callbacks for getting error and gesture recognition result.
+The enum values, as well as the parameter details, for the callback parameters are defined in the `gesture.h` and `gesture_common.h` header files.
+
+1. Error
```

Comment:
How about changing like below? I think non numbered list is proper here.

```suggestion
- Error callback
```

Would you please check?

Reply:
I changed this list. Also, I changed the list in 'Get Gesture information'.
Please check both of them. Thank you!!